File: 350. Intersection of Two Arrays II.java

import java.lang.reflect.Array;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        List<Integer> res = new ArrayList<>();

        Arrays.sort(nums1);
        Arrays.sort(nums2);

        int i = 0, j = 0, n = nums1.length, m =  nums2.length;   

        while(i < n &&j < m){
            if(nums1[i] < nums2[j]) ++i;
            else if(nums1[i] > nums2[j]) ++j;
            else{
                res.add(nums1[i]);
                ++i; ++j;
            }
        }

        return res.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 350. Intersection of Two Arrays II.cpp
#include"LeetCodeLibs.h"

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        int i1 = 0, i2 = 0;

        vector<int> v;
        while(i1 < nums1.size() && i2 < nums2.size()){
            if(nums1[i1] < nums2[i2]){ ++i1;}
            else if(nums1[i1] > nums2[i2]) ++i2;
            else {
                if(v.back() != nums1[i1])v.push_back(nums1[i1]);
                ++i1; ++i2;
            }

        }
        return v;
    }
};